### DISCUSSION TITLE
How to reset migrations to latest migration with no data in tables

### DISCUSSION THREAD

**Discussion by aderderian** [fluentmigrator/fluentmigrator#1452] [Q&A] [answered]
If I wanted to uninstall/install my data tables from scratch, any advice on how to easily do something like this? For example, we have run Fluent Migrator and are on Migration 5. The user then want's to remove all data and start the db from scratch. How could I drop all tables or rollback to a state 0, then begin running migrations from 1 forward.

Thank you!

**Comment by fubar-coder** [comment, marked as answer]
I can only see two options:

- Rollback to migration 0 and migrate to 5
  - You must not use "forward-only" migrations
- Use two "maintenance" migration: One for clearing (truncating) all tables, and another one for seeding all default values
  - The maintenance migrations should check for a flag to determine if they should be run

**Reply by jzabroski** [reply]
Correct. I use the second option and disable all constraints prior to truncating the tables.  It's important to use truncate instead of delete, as that reduces transaction log size, and development databases usually stop at 10GB.

The other trick I use with SQL Server is I use RedGate SQL Clone, but that's mostly for integration testing and to quickly undo a bad migration prior to checking it into the main branch.
